python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py

```diff
@@ -212,3 +228,107 @@ async def does_collection_exist(self, **kwargs) -> bool:
     @override
     async def delete_collection(self, **kwargs) -> None:
         await self.search_index_client.delete_index(self.collection_name, **kwargs)
+
+    @override
+    async def _inner_search(
+        self,
+        options: VectorSearchOptions,
+        search_text: str | None = None,
+        vectorizable_text: str | None = None,
+        vector: list[float | int] | None = None,
+        **kwargs: Any,
+    ) -> KernelSearchResults[Any]:
+        search_args: dict[str, Any] = {
+            "top": options.top,
+            "skip": options.skip,
+            "include_total_count": options.include_total_count,
+        }
+        if options.filter.filters:
+            search_args["filter"] = self._build_filter_string(options.filter)
+        if search_text is not None:
+            search_args["search_text"] = search_text
+        if vectorizable_text is not None:
+            search_args["vector_queries"] = [
+                VectorizableTextQuery(
+                    text=vectorizable_text,
+                    k_nearest_neighbors=options.top,
+                    fields=options.vector_field_name,
+                )
+            ]
+        if vector is not None:
+            search_args["vector_queries"] = [
+                VectorizedQuery(
+                    vector=vector,
+                    k_nearest_neighbors=options.top,
+                    fields=options.vector_field_name,
+                )
+            ]
+        if "vector_queries" not in search_args and "search_text" not in search_args:
+            # this assumes that a filter only query is asked for
+            search_args["search_text"] = "*"
+
+        if options.include_vectors:
+            search_args["select"] = ["*"]
+        else:
+            search_args["select"] = [
+                name
+                for name, field in self.data_model_definition.fields.items()
+                if not isinstance(field, VectorStoreRecordVectorField)
+            ]
+        raw_results = await self.search_client.search(**search_args)
+        return KernelSearchResults(
+            results=self._get_vector_search_results_from_results(raw_results),
+            total_count=await raw_results.get_count() if options.include_total_count else None,
+        )
+
+    def _build_filter_string(self, search_filter: VectorSearchFilter) -> str:
+        filter_string = ""
+        for filter in search_filter.filters:
+            if isinstance(filter, EqualTo):
+                filter_string += f"{filter.field_name} eq '{filter.value}' {search_filter.group_type.lower()} "
+            elif isinstance(filter, AnyTagsEqualTo):
+                filter_string += (
+                    f"{filter.field_name}/any(t: t eq '{filter.value}') {search_filter.group_type.lower()} "
+                )
+        return filter_string[:-5]
```

Comment:
Are we always assuming that the trailing string will always have 5 characters that we need to clip?

Reply:
added a check for this and a docstring.